### PR TITLE
spdx-license-list-data: 3.11 -> 3.12

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
Version bump.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116729` at 2d053326 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/6b975cc35caea6a33c00250fb6510102">esphome</a></li></li>
    <li>platformio</li>
    <li><a href="https://gist.github.com/7482c2155b2a419419ed04730266a0b4">spdx-license-list-data</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/data/misc/spdx-license-list-data/default.nix:20:3:
```
   |
20 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/data/misc/spdx-license-list-data/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by oxzi** [comment]
I have updated the PR to address the nixpkgs-hammering issues raised by @r-rmcgibbo. In this process, I set myself as this package's maintainer.

File: pkgs/data/misc/spdx-license-list-data/default.nix [hunk 1]
```diff
@@ -2,25 +2,30 @@
...
 
   src = fetchFromGitHub {
     owner = "spdx";
     repo = "license-list-data";
     rev = "v${version}";
-    sha256 = "1iwyqhh6lh51a47mhfy98zvjan8yjsvlym8qz0isx2i1zzxlj47a";
+    sha256 = "09xci8dzblg3d30jf7s43zialbcxlxly03zrkiymcvnzixg8v48f";
   };
 
   phases = [ "unpackPhase" "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/data/misc/spdx-license-list-data/default.nix:14]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/data/misc/spdx-license-list-data/default.nix (hunk 1 above)
Comment (oxzi) [reply pkgs/data/misc/spdx-license-list-data/default.nix]: I have removed this one.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>esphome</li>
<li>platformio</li>
<li>spdx-license-list-data</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
esphome:
<details>


warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/tools/misc/esphome/default.nix:1:57:

```
  |
1 | { lib, python3, platformio, esptool, git, protobuf3_12, fetchpatch }:
  |                                                         ^
```
</details>
</details>

**Pull request merged**